integration_tests/__tests__/multi_project_runner.test.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+'use strict';
+
+import runJest from '../runJest';
+import {cleanup, extractSummary, writeFiles} from '../utils';
+import os from 'os';
+import path from 'path';
+
+const skipOnWindows = require('skipOnWindows');
+const DIR = path.resolve(os.tmpdir(), 'multi_project_runner_test');
+
+skipOnWindows.suite();
+
+const fileContentWithProvidesModule = name => `/*
+ * @providesModule ${name}
+ */
+
+module.exports = {};
+`;
+
+beforeEach(() => cleanup(DIR));
+afterEach(() => cleanup(DIR));
+
+test('can pass projects or global config', () => {
+  writeFiles(DIR, {
+    '.watchmanconfig': '',
+    'package.json': '{}',
+    'project1/__tests__/file1.test.js': `
+      const file1 = require('file1');
+      test('file1', () => {});
+    `,
+    'project1/file1.js': fileContentWithProvidesModule('file1'),
```

Comment:
i used haste modules to make sure multiple roots with the same modules don't conflict with each other using MPR (happened in fb/www)